### ISSUE TITLE
[Git `HEAD`] Tests seen fail with GCC 7 and 8 — what is the oldest supported version of GCC?

### ISSUE THREAD

**Issue by hartwork** [linux-test-project/lcov#367]
With non-default GCC 7 on Ubuntu 20.04 LTS (default GCC is 9.3.0) in GitHub Actions:

```
genhtml/full.sh ................... [fail] (time 0.4s, mem 49.2MB)
    Skipping 33 previous lines (see /home/runner/work/lcov-fork/lcov-fork/tests//test.log)
    ...
      no messages were reported
    STDOUT_STOP
    STDERR_START
    Unescaped left brace in regex is deprecated here (and will be fatal in Perl 5.32), passed through in regex; marked by <-- HERE in m/({ <-- HERE lambda\(|\.lambda\$)/ at /home/runner/work/lcov-fork/lcov-fork/bin/../lib/lcovutil.pm line 4510.
    Unescaped left brace in regex is deprecated here (and will be fatal in Perl 5.32), passed through in regex; marked by <-- HERE in m/({ <-- HERE lambda\(|\.lambda\$)/ at /home/runner/work/lcov-fork/lcov-fork/bin/../lib/lcovutil.pm line 4511.
    STDERR_STOP
    Error: Output on stderr.log:
    Unescaped left brace in regex is deprecated here (and will be fatal in Perl 5.32), passed through in regex; marked by <-- HERE in m/({ <-- HERE lambda\(|\.lambda\$)/ at /home/runner/work/lcov-fork/lcov-fork/bin/../lib/lcovutil.pm line 4510.
    Unescaped left brace in regex is deprecated here (and will be fatal in Perl 5.32), passed through in regex; marked by <-- HERE in m/({ <-- HERE lambda\(|\.lambda\$)/ at /home/runner/work/lcov-fork/lcov-fork/bin/../lib/lcovutil.pm line 4511.
  EXITCODE ...: 1
make[2]: *** [../common.mak:115: check] Error 1
1 tests executed, 0 passed, 1 failed, 0 skipped (time 0.4s, mem 49.2MB)
```

**Comment by henry2cox** [comment]
GCC newer than 4.8 is expected to work - however, LLVM's gcov emulates gcov/4.2 - so it is likely that gcc 4.2 and newer should work (but I haven't tested anything older than gcc/4.8.3).

**Comment by hartwork** [comment]
@henry2cox seems to be fixed by b34c84431a97b49e55785286fa05e9a95b873920 — thanks! Also for the clarification :+1:
